server/server.go

```diff
@@ -360,6 +361,7 @@ func (s *Server) ListenAndServe() error {
 // UnixListenAndServe listens on the Unix socket address and handles keyless requests.
 func (s *Server) UnixListenAndServe() error {
```

Comment:
You should never need to call os.Remove(socket). Calling Close() on a Unix socket is what removes it.

If possible, you should Close() the socket at cleanup instead of delete it at startup, but sometimes this is unavoidable. Optimal solution is to have circus/systemd manage sockets.